### PR TITLE
Changed skills handling

### PR DESCRIPTION
**Pull request by theBGuy**
- created Skills.skill object to track our hardpoint skills, this reduces a whole lot of unneeded me.getSkill calls when we already know we have the skill
- lots of clean up and formatting

### CODE REVIEW

File: d2bs/kolbot/libs/common/Misc.js [hunk 1]
```diff
@@ -7,22 +7,2305 @@
 
 const Skill = {
 	usePvpRange: false,
-	haveTK: false,
+	skills: {
+		6: {
+			hardpoints: false,
+			checked: false,
+			have: function () {
+				if (this.hardpoints) return true;
```
Comment (noah-) [inline d2bs/kolbot/libs/common/Misc.js:15]: these have functions are largely repetitive, can you replace them with single impl where the skill is an arg?
in the cases where you have a config check, you can wrap it:

```
have: function() {
   if (condition) return false;
   return have_common();
}
```

File: d2bs/kolbot/libs/common/Pickit.js [hunk 2]
```diff
@@ -245,7 +245,7 @@ const Pickit = {
 			this.color = Pickit.itemColor(unit);
 			this.gold = unit.getStat(14);
 			this.dist = (unit.distance || Infinity);
-			this.useTk = (Skill.haveTK && Config.UseTelekinesis
+			this.useTk = (Skill.haveTK
```
Comment (noah-) [inline d2bs/kolbot/libs/common/Pickit.js:248]: should retain a way to disable TK

Comment (theBGuy) [reply d2bs/kolbot/libs/common/Pickit.js:248]: This does still have a way to disable it, Config.Telekinesis is handled in the Skill.skills object

File: d2bs/kolbot/libs/common/Misc.js (hunk 1 above)
Comment (theBGuy) [reply d2bs/kolbot/libs/common/Misc.js]: Done, that cleaned it up a lot

**Pull request merged**